.github/workflows/release.yml

```diff
@@ -0,0 +1,105 @@
+name: Build distribution
+on:
+  release:
+    types:
+      - published
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build-artifacts:
+    runs-on: ubuntu-latest
+    if: github.repository == 'zarr-developers/VirtualiZarr'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v5.0.0
+        name: Install Python
+        with:
+          python-version: "3.12"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install setuptools setuptools-scm wheel twine check-manifest
+          git clean -xdf
+          git restore -SW .
+      # This step is only necessary for testing purposes and for TestPyPI
+      - name: Fix up version string for TestPyPI
+        if: ${{ !startsWith(github.ref, 'refs/tags') }}
+        run: |
+          # Change setuptools-scm local_scheme to "no-local-version" so the
+          # local part of the version isn't included, making the version string
+          # compatible with PyPI.
+          sed --in-place "s/guess-next-dev/no-local-version/g" pyproject.toml
+      - name: Build tarball and wheels
+        run: |
+          python -m build --sdist --wheel .
+      - name: Check built artifacts
+        run: |
+          python -m twine check --strict dist/*
+          pwd
+          if [ -f dist/virtualizarr-unknown.tar.gz ]; then
+            echo "❌ INVALID VERSION NUMBER"
+            exit 1
+          else
+            echo "✅ Looks good"
+          fi
+      - uses: actions/upload-artifact@v4
+        with:
+          name: releases
+          path: dist
+
+  test-built-dist:
+    needs: build-artifacts
+    runs-on: ubuntu-latest
+    environment:
+      name: test-release
+      url: https://test.pypi.org/p/virtualizarr
```

Comment:
Have you reserved the name on pypi? Or will it do that the first time this is run?

Reply:
That's totally fine - yes please add me to reduce the bus factor (TomNicholas) on pypi. Will anyone with the ability to push a tag to the repo be able to release?

Reply:
I'm going to add a super simple contributing guide that goes over this structure so it's easier to find these answers in the future. To make that, are you planning to use SemVer or year.month versioning for the tags/releases?

Reply:
Great idea! 

> are you planning to use SemVer or year.month versioning

I prefer SemVer (or [EffVer](https://jacobtomlinson.dev/effver/)). I don't expect this package to have so many ongoing changes that it needs really frequent releases indefinitely into the future - the scope is pretty targeted so I'm hoping it converges to some stable fixed point, with only bugfix releases after that. (We're a fair way from that now though.)

Reply:
@TomNicholas can you let me know if https://virtualizarr--136.org.readthedocs.build/en/136/contributing.html#making-a-release answers your questions?

Reply:
This is great! It's almost exactly the same as [xarray's release guide](https://github.com/pydata/xarray/blob/main/HOW_TO_RELEASE.md), which is blessedly simple :pray:

I think the only differences are:
1. The need to go and manually approve the deployment
2. Xarray also publishes to conda-forge at the same time

Reply:
1. We could change this so there aren't any environment protection rules if you rather
2. True, I'd be glad to set up the conda-forge package which would also be automatic without any more changes needed via GitHub actions. It's easier to do it after it's on PyPI because then you can just use [grayskull](https://pypi.org/project/grayskull/).

Reply:
> We could change this so there aren't any environment protection rules if you rather

This already requires the would-be-releaser to have permissions to create tags right? Which is already restricted to `virtualizarr-core-devs` I believe?

> True, I'd be glad to set up the conda-forge package which would also be automatic without any more changes needed via GitHub actions. It's easier to do it after it's on PyPI because then you can more easily use [grayskull](https://pypi.org/project/grayskull/).

That would also be great. Also I love the name of that package.

Reply:
This already requires the would-be-releaser to have permissions to create tags right? Which is already restricted to virtualizarr-core-devs I believe?

AFAIK anyone with write permissions can create tags. It's possible to restrict tags beyond that to admins (i.e., virtualizarr-core-devs) via protected tags (https://github.com/zarr-developers/VirtualiZarr/settings/tag_protection). I'm not sure who in the zarr-developers organization has write permissions to all repos but not admin permissions, which is who that restriction could apply to. For now, I've just removed the workflow protection and will merge the PR.